src/profile/notification-settings/NotificationSettings.ts

```diff
@@ -0,0 +1,184 @@
+import _ from 'lodash/fp';
+import { useEffect, useState } from 'react';
+import { div, h, h2 } from 'react-hyperscript-helpers';
+import { spinnerOverlay } from 'src/components/common';
+import { InfoBox } from 'src/components/InfoBox';
+import { PageBox, PageBoxVariants } from 'src/components/PageBox';
+import { memoWithName } from 'src/libs/react-utils';
+import { userStore } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import {
+  NotificationCard,
+  NotificationCardProps,
+  UserAttributesCard,
+  UserAttributesCardProps,
+} from 'src/profile/notification-settings/PreferenceCard';
+import {
+  workspaceChangedNotificationInfo,
+  workspaceChangedNotificationKey,
+  workspaceSubmissionNotificationInfo,
+  workspaceSubmissionNotificationKeys,
+} from 'src/profile/notification-settings/utils';
+import { useWorkspaces } from 'src/workspaces/common/state/useWorkspaces';
+import { WorkspaceWrapper } from 'src/workspaces/utils';
+
+// "Headers" for each section (note that checkboxes have aria labels to make up for the fact that the information
+// is not actually being displayed in a table).
+const headerGridStyle = { display: 'grid', gridTemplateColumns: 'repeat(4, 1fr)', margin: '1.5rem 0 0.5rem 0' };
```

Comment:
I changed to a grid layout for consistent layout between the sections (given that the last section will now have columns of checkboxes).

Reply:
Would it be _better_ to use a table? This seems to count as tabular data, to me... 🤷 

I'm not sure what kind of global styles we have set up on tables though. Sometimes those can make them more trouble than they're worth.

Reply:
I'm not sure a table would be better in this case. Having controls within a table is a bit wonky, and our tables aren't "real" tables anyway. Right now these are lists.